internal/transport/transport.go

```diff
@@ -457,9 +457,10 @@ func (s *Stream) Read(p []byte) (n int, err error) {
 // the stream broke.
 type transportReader struct {
 	reader io.Reader
+	stream *Stream
```

Comment:
not sure if you'll be happy with this, it's a bit janky

Reply:
Can you measure the allocations saved by this optimization?  It seems it might not be saving as much as you're expecting because of the need to store the stream pointer, too.

Reply:
dang, i think you may be right. i didn't measure this bit and i bet it doesn't help that much, but the other closure (where we had Stream already) should be pure win. i'll try to measure when i get a chance